event-handlers/src/main/java/no/sikt/nva/nvi/events/evaluator/calculator/CreatorVerificationUtil.java

```diff
@@ -134,32 +145,33 @@ private List<NviOrganization> getNviAffiliationsIfExist(JsonNode contributorNode
         return streamNode(contributorNode.at(JSON_PTR_AFFILIATIONS))
                    .map(JsonUtils::extractId)
                    .distinct()
-                   .map(organizationRetriever::fetchOrganization)
-                   .filter(this::topLevelOrgIsNviInstitution)
+                   .parallel()
+                   .map(this::fetchOrgAsync)
+                   .map(CreatorVerificationUtil::getFuture)
+                   .filter(organization -> topLevelOrgIsNviInstitutionAsync(organization).join())
                    .map(CreatorVerificationUtil::toNviOrganization)
                    .toList();
     }
 
-    private boolean topLevelOrgIsNviInstitution(Organization organization) {
-        return isNviInstitution(organization.getTopLevelOrg().id());
+    private CompletableFuture<Boolean> topLevelOrgIsNviInstitutionAsync(Organization organization) {
+        var customerApiUri = createCustomerApiUri(organization.getTopLevelOrg().id().toString());
+        return CompletableFuture.supplyAsync(() -> getResponse(customerApiUri))
+                   .thenApply(response -> {
+                       if (isSuccessOrNotFound(response)) {
+                           return mapToNviInstitutionValue(response);
+                       } else {
+                           LOGGER.error(COULD_NOT_FETCH_CUSTOMER_MESSAGE + customerApiUri + ". Response code: {}",
+                                        response.statusCode());
+                           throw new RuntimeException(COULD_NOT_FETCH_CUSTOMER_MESSAGE + customerApiUri);
+                       }
+                   });
     }
 
-    private boolean isNviInstitution(URI institutionId) {
-        var customerApiUri = createCustomerApiUri(institutionId.toString());
-        var response = getResponse(customerApiUri);
-        if (isSuccessOrNotFound(response)) {
-            return mapToNviInstitutionValue(response);
-        }
-        LOGGER.error(COULD_NOT_FETCH_CUSTOMER_MESSAGE + customerApiUri + ". Response code: {}", response.statusCode());
-        throw new RuntimeException(COULD_NOT_FETCH_CUSTOMER_MESSAGE + institutionId);
+    private CompletableFuture<Organization> fetchOrgAsync(URI uri) {
+        return CompletableFuture.supplyAsync(() -> organizationRetriever.fetchOrganization(uri));
     }
 
     private HttpResponse<String> getResponse(URI uri) {
-        return Optional.ofNullable(authorizedBackendUriRetriever.fetchResponse(uri, CONTENT_TYPE))
```

Comment:
I did not see the value in `Optional<Optional<Something>>` here, so i refactored this method